Scrapper-Service/plugins/guide2research.py

```diff
@@ -0,0 +1,354 @@
+                     category_c: category_c_value,
+                     "g_scholar_h5_index": hindex})
+        except Exception as e:
+            self.logger.error(
+                f"Error occured when generating ranking information for {name}, full error {e}")
+
+    def _get_top_conf_bulk_text(self, soup: bs) -> str:
+        """
+        Parses page soup and returns bulk text
+
+        Args:
+        ---
+        soup: bs4 soup
+
+        Returns:
+        ---
+        bulk text: str
+        """
+        try:
+            paragraphs = soup.find_all("p")
+            bulk_text = "" 
+            for p in paragraphs:
+                bulk_text+= p.text.strip()
+            return bulk_text
+        except Exception as e:
+            self.logger.error(f"Error parsing bulk text{e}")
+
+    def _parse_all_conference_base(self, content: bytes) -> List[str]:
+        soup = bs(content, "html5lib")
+        conference_div = soup.find("div", attrs={"id": "ajax_content"})
+        conf_tables = conference_div.findAll("table", attrs={"cellspacing": "0"})
+        links = []
+        try:
+            for conf in conf_tables:
+                details = conf.findAll("td")
+                anchor = details[1].h4.a
+                if anchor:
+                    link = self.base_address + anchor['href']
+                    links.append(link)
+                else:
+                    self.logger.error(f'Conference link not found')
+            return links
+        except Exception as e:
+            self.logger.error(f"Failed to parse links in all conferences page with error {e}")
+
+    def _parse_all_conference(self, link: str):
+        """
+        Parses individual conference page and,
+        return Conference object
+
+        Args:
+        ---
+        link: str
+
+        Returns:
+        ---
+        Conference: object
+        """
+        self.logger.info(f'Parsing {link}')
+        page = requests.get(link, allow_redirects=True)
+        if not page.status_code == 200:
+            self.logger.error(f"loading {link} failed with {page.status_code}")
+        try:
+            content = page.content
+        except Exception as e:
+            self.logger.error(
+                f"The following error occured while trying to parse {link} {e}")
+        soup = bs(content, "html5lib")
+        content_div = soup.find("div", attrs={"id": "content_box"})
+        title = content_div.h1.text
+        tables = soup.find_all('table')
+        conf_info = self._get_all_conf_info(infotable=tables[0])
+        bulk_text = self._get_all_conf_bulk(soup=soup)
+        metadata = Metadata(__name__,
+                            dt.now(),
+                            link,
+                            self.base_address,
+                            'Guide2Research',
+                            )
+        additional_data = {}
+        additional_data["bulkText"] = bulk_text
+        additional_data["dates"] = conf_info.get("dates")
```

Comment:
* The field information exist with a different name already  (dates -> dateRange , address -> location)  in the database. look into the optional arguments of Conference Model.
* Match those, so that there will be no replication of data. Check in the conference.py datamodel for the optional fields.

Reply:
Done